### ISSUE TITLE
The options

### ISSUE THREAD

**Issue by Skamer** [Skamer/Syling-Tracker#20]
Hello everyone, 

The options are a hot topic so i have decided to open a ticket for having a space where we could discuss about them and be notified on their progress. 

I'd like also to get your wishes at this purpose. What are the things you'd like to have inside ? Please to put them here.

**Comment by RyadaProductions** [comment]
For me I only really have a few small options that i would like to be added as a toggle:

An option to hide quests/achievements etc in combat while inside of a raid.
As an addition to the above hide everything except the keystone information when a M+ key has been started.

**Comment by AmenitradieDruidin** [comment]
Oh I saw this too late. I have the following wishes:

- the search option for groups in world boss / world quest
- Hide tracker during pet fight

**Comment by Skamer** [comment]
Thanks for your posts. 

---

### The categories

I understand you prefer to not have the categories and something more compact. Good news, the addon already supports well the both, it's just the displaying of categories is set to true  as default. Once i finished the 0.4 version which will implement the Torghast support, i'll consider adding a slash command (/slt categories [true/false]) to enable or disable them. 

![image](https://user-images.githubusercontent.com/849450/97791455-d1179d80-1bd2-11eb-8a43-8667168313b6.png)

---

### The Search options 

It's an oversight from my part, the World quest should be provide a context menu with Search function. I have already noticed some issue with the Search function where there are Taint errors (reference to #21 ).  I haven't yet the opportunity to look in detail on why this happens.

**Comment by raol12ad** [comment]
Some things i'm interested in:

- Option to hide the minimap button, it doesn't even seem to do anything for me, so i have no idea why it is there :P?
- Option to change transparency of the background of the top bar as well as the quests
- Minimize button on the frame, so you can minimize it all into a single box when you dont want to see the quests.
- Option to auto-minimize quest tracker in combat / hide it entirely (could also be an option to hide it when in raids / bgs etc.)

**Comment by Skamer** [comment]
> Option to hide the minimap button, it doesn't even seem to do anything for me, so i have no idea why it is there :P?

Currently, the minimap button does nothing except for displaying the addon version (probably the reason on why i have kept it). In the future, it will be more useful as you will open the options from it.

---
I added some new slash commands in the version 0.4 which is now available.

- `/slt minimap` -- to toggle the minimap icon.
- `/slt qcat` -- to toggle the displaying of categories for quests. Take note, the change will take effect after the next data update or a reload ui. You can moving your character or opening your quest log to trigger a data update so the change is applied.

--- 
## Interaction with WeakAuras 2

I'm experimenting an interesting interaction between Syling Tracker and WeakAuras2. I use this one to pilot the addon on certain stuffs.

### Example 1: Create an auto hide/auto shown

Here, when my player has power shield buff on him, Torghast will be displayed. Once the buff goes away, Torghast won't be shown.

![slt_weakaura_track_torghast_example](https://user-images.githubusercontent.com/849450/98064087-c454ad00-1e51-11eb-97ba-9f8c12295638.gif)

**How this works:**

In the 0.4 version, i did some change to provide a public way for tracking or untracking a content type. 

This function will untrack Torghast. My WeakAura calls this function in Actions/OnHide/Custom.
```lua
Scorpio.FireSystemEvent("SLT_UNTRACK_CONTENT_TYPE", "torghast")
```

This function will track Torghast. My WeakAura calls this function in Actions/OnShow/Custom.
```lua
Scorpio.FireSystemEvent("SLT_TRACK_CONTENT_TYPE", "torghast")
```

The both functions needs as first arg, the content type you want tracker or untrack. The value possible are the following:
```lua
"scenario"
"dungeon"
"achievements"
"bonus-tasks"
"tasks"
"quests"
"auto-quests"
"world-quests"
"keystone"
"torghast"
```

### Example 2: Changing the visual of the addon. 

An another interesting thing we can do is to change the visual of the addon. In the bellow example, i change the font size of quest name, and remove their background.

![image](https://user-images.githubusercontent.com/849450/98065594-295dd200-1e55-11eb-8d21-799d77505dbf.png)

**How this works**

Syling Tracker uses a system skin which rules all the addon visual (nothing is hard coded so can be easily customized). In this example, i create a WeakAuras which create a skin definition and apply it to Syling Tracker.

---

**Comment by SaturosZed** [comment]
I'm really looking forward to the options for changing fonts and colors so it fits the theme of my UI.
![WoWScrnShot_110820_084138](https://user-images.githubusercontent.com/42712086/98459812-9f886e80-219e-11eb-8c82-803ac1fc273b.jpg)

**Comment by Skamer** [comment]
I'm starting the preparation of a long work on the options. I have finalized the main design and how the navigation will be done between different sections.

<details> 
  <summary>Here a very very early preview</summary>

![slt_options_nav_early_preview](https://user-images.githubusercontent.com/849450/98487266-4eba5980-2222-11eb-92e6-7f31470a8aa9.gif)
![slt_options_early_preview_](https://user-images.githubusercontent.com/849450/98487244-2fbbc780-2222-11eb-8c91-c6a715dd5a97.jpg)
</details>

As you can notice it, the visual is very different compared to what you can see on another addons. I have wanted something new and more visually pleasing.

**Comment by RyadaProductions** [comment]
After a bit of thinking the ability to toggle the tracker on/off based on specific things you can specify.  I know I suggested this before but thought it would be nice to have a few examples of the triggers.
Some examples of options where it could be nice to be able to toggle it on or off that I can think of:
- When in a Battleground.
- When in an Arena.
- While in combat in a Raid
- While on a flight path

**Comment by Skamer** [comment]
This is a very long time since my last post. 

With Dragonflight is coming, i plan to come back on WoW, as a consequence the addon will get update again.

I relaunch the development of options.

Todo List:

Step 1: Create the widgets needed for options
_The widgets are based on the new Blizzard style UI._
- [ ] DropDown
- [x] Window
- [x] Panel
- [x] Category
- [ ] CheckBox
- [ ] Tabs
- [ ] EditBox
- [ ] Expandable Section 

Step 2: Create the option panel
- [ ] General Category
  -  [ ] Item Bar
  -  [ ] Profiles
- [ ] My Trackers Category
- [ ] Contents Category
  -  [ ] Quests (includes the normal quests, dungeon and raid quests).
  -  [ ] Campaign (includes the campaign quests)
  -  [ ] World quests
  -  [ ] Dungeon
  -  [ ] Keystone (mythic +)
  -  [ ] Achievements
  -  [ ] Bonus Tasks
  -  [ ] Task
  -  [ ] Torghast

Step 3: Implement and bind the options.

**Comment by Skamer** [comment]
Hello,
With the prepatch is coming, i have decided to change my initial plans as i'm unable to provide you a version ready and stable with all i have planned for the prepatch release.

My initial plan was to release it once all is ready. Unfortunately doing that will again add delay, and to be honest i'd like you have the options the more quickly as possible, so instead to release them all in once, i decided to do by iteration.

I'll soon begin the version will be published once the prepatch hits the server live. It will prepare the integration of options, clicking the addon on the minimap or doing a /slt config will load and open the option panel. Take note, only few options will be available. Nevertheless you can expect from this version each new ones will add new entries in the options panel, so don't hesitate to check the changelog for seeing what is added.

**Comment by Skamer** [comment]
The first iteration of options has been added in the 1.0.0 is now available

**Comment by Skamer** [comment]
I close the ticket, but i don't forget the requests asked.
